### PR TITLE
sort samples in batch-edit by sample code

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

**Review by Steffengreiner — DISMISSED** [review]
Perfect 👌

**Review by Steffengreiner — CHANGES_REQUESTED** [review]
Although the code is perfect, I'd request to change the target branch to dev 😅

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationMain.java [hunk 1]
```diff
@@ -264,8 +265,10 @@ private void onEditBatchClicked(EditBatchEvent editBatchEvent) {
         editBatchEvent.batchPreview().batchId()).stream().toList();
     var experimentalGroups = experimentInformationService.experimentalGroupsFor(
         context.experimentId().get());
-    List<SampleBatchInformationSpreadsheet.SampleInfo> sampleInfos = samples.stream()
-        .map(sample -> convertSampleToSampleInfo(sample, experimentalGroups)).toList();
+    // need to create mutable list to order samples
+    List<SampleBatchInformationSpreadsheet.SampleInfo> sampleInfos = new ArrayList<>(samples.stream()
+        .map(sample -> convertSampleToSampleInfo(sample, experimentalGroups)).toList());
```
Comment (KochTobi) [inline user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationMain.java:270]:
```suggestion
    List<SampleBatchInformationSpreadsheet.SampleInfo> sampleInfos = samples.stream()
        .map(sample -> convertSampleToSampleInfo(sample, experimentalGroups))
        .collect(Collectors.toCollection(ArrayList::new));
```

Comment (KochTobi) [reply user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationMain.java:270]: du kannst auch den stream schon sorten
```suggestion
    var sampleInfos = samples.stream()
        .map(sample -> convertSampleToSampleInfo(sample, experimentalGroups))
        .sorted(Comparator.comparing(o -> o.getSampleCode().code())
        .collect(Collectors.toCollection(ArrayList::new));
```

**Review by Steffengreiner — APPROVED** [review]
Send it! ✉️

**Pull request merged**